### ISSUE TITLE
Saving agent weights during the training

### ISSUE THREAD

**Issue by ViktorM** [matthiasplappert/keras-rl#94]
What is a canonical way to tell the Keras-RL agent (DDPG for example) to save weights after epochs with the best reward? And to regularly save weights, for example after each 5 training epochs? Is it currently supported?

**Comment by matthiasplappert** [comment]
Check out `rl/callbacks.py` and in particular `ModelIntervalCheckpoint`. This allows you to save checkpoints every x steps. The DQN Atari example (`examples/dqn_atari.py`) illustrates how you can use it. The callback should work for every agent in Keras-RL.

Regarding saving the best weights: You should be able to easily write you own callback for this purpose. You can also check out `ModelCheckpoint` (https://github.com/fchollet/keras/blob/master/keras/callbacks.py#L314) from Keras, which should also work.
